cmd/boulder/main_test.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/test"
+)
+
+// TestConfigValidation checks that each of the components which register a
+// validation tagged Config struct at init time can be used to successfully
+// validate their corresponding test configuration files.
+func TestConfigValidation(t *testing.T) {
+	// The list of components to test. Each component is a pair of at lease one
+	// `test/config` file path and a `cmd` package name.
+	components := make(map[string][]string)
+
+	configPath := "../../test/config"
+	if os.Getenv("BOULDER_CONFIG_DIR") == "test/config-next" {
+		configPath = "../../test/config-next"
+	}
```

Comment:
Move this to the top of the function, so it's not separating the declaration of the `components` map from the loop that populates that map.

Or move it to after the loop, since I also suggest only using this value inside the lower loop, not inside the switch/case.